subsystems/viamserver/viamserver.go

```diff
@@ -294,17 +297,80 @@ func (s *viamServer) HealthCheck(ctx context.Context) (errRet error) {
 	return errRet
 }
 
+func (s *viamServer) isRestartAllowed(ctx context.Context) (restartAllowed bool, errRet error) {
```

Comment:
Should comment that this needs to be called only from inside a lock (since it uses s.checkURL/s.checkURLAlt fields from s.)

Reply:
Good point; done.